Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs

```diff
@@ -0,0 +1,153 @@
+
+    /// <summary>
+    /// The base image tag.
+    /// Ex: 6.0
+    /// </summary>
+    [Required]
+    public string BaseImageTag { get; set; }
+
+    /// <summary>
+    /// The registry to push to.
+    /// </summary>
+    [Required]
+    public string OutputRegistry { get; set; }
+
+    /// <summary>
+    /// The name of the output image that will be pushed to the registry.
+    /// </summary>
+    [Required]
+    public string ImageName { get; set; }
+
+    /// <summary>
+    /// The tag to associate with the new image.
+    /// </summary>
+    public ITaskItem[] ImageTags { get; set; }
+
+    /// <summary>
+    /// The directory for the build outputs to be published.
+    /// Constructed from "$(MSBuildProjectDirectory)\$(PublishDir)"
+    /// </summary>
+    [Required]
+    public string PublishDirectory { get; set; }
+
+    /// <summary>
+    /// The working directory of the container.
+    /// </summary>
+    [Required]
+    public string WorkingDirectory { get; set; }
+
+    /// <summary>
+    /// The entrypoint application of the container.
+    /// </summary>
+    [Required]
+    public ITaskItem[] Entrypoint { get; set; }
+
+    /// <summary>
+    /// Arguments to pass alongside Entrypoint.
+    /// </summary>
+    public ITaskItem[] EntrypointArgs { get; set; }
+
+    /// <summary>
+    /// Labels that the image configuration will include in metadata
+    /// </summary>
+    public ITaskItem[] Labels { get; set; }
+
+    /// <summary>
+    /// Ports that the application declares that it will use.
+    /// Note that this means nothing to container hosts, by default -
+    /// it's mostly documentation.
+    /// </summary>
+    public ITaskItem[] ExposedPorts { get; set; }
+ 
+    // Unused, ToolExe is set via targets and overrides this.
+    protected override string ToolName => "dotnet";
+
+    private string DotNetPath
+    {
+        get
+        {
+            string path = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") ?? "";
+            if (string.IsNullOrEmpty(path))
+            {
+                path = string.IsNullOrEmpty(ToolPath) ? "" : ToolPath;
+            }
+
+            return path;
+        }
+    }
+
+    public CreateNewImage()
+    {
+        ContainerizeDirectory = "";
+        BaseRegistry = "";
+        BaseImageName = "";
+        BaseImageTag = "";
+        OutputRegistry = "";
+        ImageName = "";
+        ImageTags = Array.Empty<ITaskItem>();
+        PublishDirectory = "";
+        WorkingDirectory = "";
+        Entrypoint = Array.Empty<ITaskItem>();
+        EntrypointArgs = Array.Empty<ITaskItem>();
+        Labels = Array.Empty<ITaskItem>();
+        ExposedPorts = Array.Empty<ITaskItem>();
+    }
+
+    protected override string GenerateFullPathToTool() => Quote(Path.Combine(DotNetPath, ToolExe));
+
+    protected override string GenerateCommandLineCommands()
+    {
+        return Quote(ContainerizeDirectory + "containerize.dll") + " " +
```

Comment:
For later refactor: this should use `Microsoft.Build.Utilities.CommandLineBuilder` (and @baronfel, we should probably expend some effort to make sure `CommandLineBuilder` plays well with S.CL).

Reply:
https://github.com/dotnet/sdk-container-builds/issues/172